vanilla/vanilla
# Add functions for comparing ranked permissions

Partially addresses vanilla/support#1571

This PR adds functions that allow comparison between the current user's permissions and the permissions of a user the current user wants to perform some kind of administrative action on (ban, mark as a troll, etc.). These functions can be called by other classes to make sure a user isn't able to take an action against another user with the same or higher permissions.

The PR also implements this check in the ban method of the `usercontroller.php`

### TO TEST
1. Make sure the moderator role can warn users (enable `Garden.Moderation.Manage` for that role).
1. Make sure you have at least to moderator users.
1. As one moderator, try to ban the other. Verify that it works.
1. Check out this branch and repeat step 4, but verify that it doesn't work.
